content/waku/2024-gantt.md

```diff
@@ -0,0 +1,104 @@
+        Review MVDS usage and fail path: crit, milestone, after mvds, 0
+    section Static Sharding - dedicated shards
+        (telemetry) Measure Bandwidth: milestone, after telem-bandwidth, 0
+        Sharding peer mgmt and discovery hardening: crit, milestone, after sh-peer-mgmt-j sh-peer-mgmt-n sh-peer-mgmt-g, 0
+        (testing) Custom shard impl of Communities: milestone, after test-custom-shard, 0
+
+%% Tasks
+    section golang eng 1
+        %% Estimation TBC - Prem says fine, waiting on 2nd opinion
+        Review connection management: crit, review-conn-mgmt-1, 2024-05-13, 8w
+        %% Estimation TBC - Prem says fine, waiting on 2nd opinion
+        Review MVDS usage and fail path: crit, mvds, after review-conn-mgmt-1, 6w
+        %% Estimate/start TBC
+        (testing) direct messages: test-direct-msg, after mvds, 4w
+        %% Estimate TBC
+        (testing) Custom shard impl of Communities: test-custom-shard, after test-direct-msg, 4w 
+    section golang eng 2
+        Reliability Protocol for Relay (status-go): crit, rel-relay-g, 2024-05-13, 12w
+    section golang eng 3
+        Store v3 (go-waku client only): crit, storev3-g, 2024-02-26, 2024-05-24
+        %% Estimate TBC - assuming parallel work possible
+        Review connection management: crit, review-conn-mgmt-2, after storev3-g, 8w
+        Sharding peer mgmt and discovery hardening (go-waku): crit, sh-peer-mgmt-g, after review-conn-mgmt-2, 8w
+    section golang eng 4
+        %% Review if actually needed
+        Store v3 (sync): crit, 2024-02-08, 2024-04-26
+        Reliability Protocol for Resource-Restricted Clients (status-go): crit, rel-reqres-g, 2024-05-13, 10w
+        (metric) Count store messages: count-store-msg, after rel-reqres-g, 2w
+    section golang eng 5
+        %% Estimate TBD - Prem says fine, waiting on 2nd opinion
+        DoS protection for req-res protocols (go-waku client only): crit, dosreqresg, 2024-05-20, 4w
+        %% TBC estimate and start
+        (telemetry) direct message reliability: telem-d-msg-rel, 2024-06-17, 6w
+        %% TBC estimate and start
+        (telemetry) Measure Bandwidth: telem-bandwidth, after telem-d-msg-rel, 8w
+    section test eng 1
+        %% TBC estimate
+        %% This is Store Service Upgrade - item (1) in DST simulation - start with small scale to get faster results 
+        (simulation) Stress test store v3: sim-storev3, 2024-05-13, 4w
+        %% TBC estimate
+        %% This is Store Service Upgrade - item (2) in DST simulation - start with small scale to get faster results
+        (simulation) Compare store topologies: sim-store-cmp, after sim-storev3, 6w
+        %% TBC estimate/priority
+        (simulation) req-res protocol performance: sim-req-res, after sim-store-cmp, 6w
+        %% TBC estimate
+        (simulation) relay reliability performance impact: sim-relay-rel, after sim-req-res rel-relay-g rel-relay-n, 4w
+        %% TBC estimate
```

Comment:
@jm-clius WDYT?

I am not sure that it would fall @stubbsta's mandate to do some peer/connection management exercises?

What about simulations in terms of discovery in a sharded environment?

Any other items it would be useful to do before DST can?

Reply:
Also, what about connection dropping etc? could we produce preliminary results with our tooling?

Reply:
Yes, I would say the following is useful and doable within smaller scale simulations:
- peer and connection management tests (can set up rough roadmap and deliverables here)
- store v3 full functionality and capacity test (e.g. number of message hash "pings" per min, store query performance, etc.)

Discovery in a sharded environment would roughly be covered in "peer and connection management" in the short term. In the longer term (e.g. testing max number of shards, or scenarios of > 100 communities), @stubbsta would likely be involved in collaboration with the discv5/rendezvous evaluation effort.

Reply:
Update. @jm-clius let me know how it looks in terms of estimate.

At the moment it's scheduled one simulation after the other but I assume that it's going to be more in parallel in execution. ie, setup a simulation for peer mgmt, run it, feedback, then do next task while bugs are fixed.